src/container/srv_container.c

```diff
@@ -2410,37 +2440,72 @@ cont_prop_read(struct rdb_tx *tx, struct cont *cont, uint64_t bits,
 		d_iov_set(&value, &val, sizeof(val));
 		rc = rdb_tx_lookup(tx, &cont->c_prop, &ds_cont_prop_ec_pda,
 				   &value);
-		if (rc == -DER_NONEXIST) {
-			rc = 0;
+		if (rc == -DER_NONEXIST)
 			val = DAOS_PROP_PO_EC_PDA_DEFAULT;
-		} else  if (rc != 0) {
+		else if (rc != 0)
 			D_GOTO(out, rc);
-		}
 		D_ASSERT(idx < nr);
 		prop->dpp_entries[idx].dpe_type = DAOS_PROP_CO_EC_PDA;
 		prop->dpp_entries[idx].dpe_val = val;
+		if (rc == -DER_NONEXIST) {
+			prop->dpp_entries[idx].dpe_flags |= DAOS_PROP_ENTRY_NEGATIVE;
```

Comment:
Looking at the usage of this `DAOS_PROP_ENTRY_NEGATIVE` flag, might be better to name it something like `DAOS_PROP_ENTRY_NOT_SET`